src/main/java/org/fcrepo/spec/testsuite/crud/StateToken.java

```diff
@@ -0,0 +1,198 @@
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.spec.testsuite.crud;
+
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertNotNull;
+
+import io.restassured.http.Header;
+import io.restassured.http.Headers;
+import io.restassured.response.Response;
+import org.fcrepo.spec.testsuite.AbstractTest;
+import org.fcrepo.spec.testsuite.TestInfo;
+import org.testng.SkipException;
+import org.testng.annotations.Test;
+
+/**
+ * @author dbernstein
+ */
+public class StateToken extends AbstractTest {
+    private static final String STATE_TOKEN = "X-State-Token";
+    private static final String IF_STATE_TOKEN = "X-If-State-Token";
+
+    /**
+     * 3.10.1-A
+     */
+    @Test(groups = {"MAY"})
+    public void stateTokenGet() {
+        final TestInfo info = setupTest("3.10.1-A",
+                                        "Implementations may include the X-State-Token header field in GET responses " +
+                                        "to provide a token representing the current state of resource. If provided, " +
+                                        "this value must change whenever the underlying state of the resource has " +
+                                        "changed.",
+                                        "https://fcrepo.github.io/fcrepo-specification/#x-state-token", ps);
+
+        final Response createResponse = createBasicContainer(uri, info.getId());
+        final String resourceUri = getLocation(createResponse);
+        final Response getResponse = doGet(resourceUri);
+        checkStateTokensSupported(getResponse);
+        final String value = getResponse.getHeader(STATE_TOKEN);
+        assertNotNull(STATE_TOKEN + " must not be null.", value);
+        final String sparqlUpdate =
+            "INSERT { <> <http://purl.org/dc/elements/1.1/title> \"test\" . } WHERE {}";
+        doPatch(resourceUri, new Headers(new Header("Content-Type", "application/sparql-update")), sparqlUpdate);
+        assertEquals("New " + STATE_TOKEN + "should not match previous token value", value,
+                     doGet(resourceUri).header(STATE_TOKEN));
+    }
+
+    private void checkStateTokensSupported(final Response response) throws SkipException {
+        if (!isStateTokenSupported(response)) {
+            throw new SkipException("This implementation does not support " + STATE_TOKEN);
+        }
+
+        return;
```

Comment:
`return` at end of `void` method is redundant.

Reply:
👍